python/glow/wgr/linear_model/logistic_model.py

```diff
@@ -0,0 +1,238 @@
+                logistic_reduced_matrix_struct, PandasUDFType.GROUPED_MAP)
+            join_type = 'right'
+        else:
+            raise ValueError(f'response must be either "linear" or "sigmoid", received "{response}"')
+
+        return blockdf.drop('header_block', 'sort_key') \
+            .join(modeldf.drop('header_block'), ['sample_block', 'header'], join_type) \
+            .withColumn('label', f.coalesce(f.col('label'), f.col('labels').getItem(0))) \
+            .groupBy(transform_key_pattern) \
+            .apply(transform_udf) \
+            .join(cvdf, ['label', 'alpha'], 'inner')
+
+    def transform(self, blockdf: DataFrame, labeldf: pd.DataFrame, sample_blocks: Dict[str,
+                                                                                       List[str]],
+                  modeldf: DataFrame, cvdf: DataFrame) -> pd.DataFrame:
+        """
+        Generates GWAS covariates for the target labels in the provided label DataFrame by applying the model resulting
+        from the LogisticRegression fit method to the starting block matrix.
+
+        Args:
+            blockdf : Spark DataFrame representing the beginning block matrix X
+            labeldf : Pandas DataFrame containing the target labels used in fitting the ridge models
+            sample_blocks : Dict containing a mapping of sample_block ID to a list of corresponding sample IDs
+            modeldf : Spark DataFrame produced by the LogisticRegression fit method, representing the reducer model
+            cvdf : Spark DataFrame produced by the LogisticRegression fit method, containing the results of the cross
+            validation routine.
+
+        Returns:
+            Pandas DataFrame containing  covariate values. The shape and order match labeldf such that the
+            rows are indexed by sample ID and the columns by label. The column types are float64.
+        """
+        block_prediction_df = self.reduce_block_matrix(blockdf,
+                                                       labeldf,
+                                                       sample_blocks,
+                                                       modeldf,
+                                                       cvdf,
+                                                       covdf=pd.DataFrame({}),
+                                                       response='linear')
+        pivoted_df = flatten_prediction_df(block_prediction_df, sample_blocks, labeldf)
+
+        record_hls_event('wgrLogisticRegressionTransform')
+
+        return pivoted_df
+
+    def predict_proba(self, blockdf: DataFrame, labeldf: pd.DataFrame,
+                      sample_blocks: Dict[str, List[str]], modeldf: DataFrame, cvdf: DataFrame,
+                      covdf: pd.DataFrame) -> pd.DataFrame:
+        """
+        Generates predicted probabilities for the target labels in the provided label DataFrame by applying the model resulting
```

Comment:
I think we need some extra words in this vs `transform` to explain the difference. Currently the descriptions are identical.